src/internal_modules/roc_core/seqlock_impl.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "roc_core/seqlock_impl.h"
+
+#include "roc_core/atomic_ops.h"
+#include "roc_core/cpu_instructions.h"
```

Comment:
nit: extra empty line.

Reply:
Oh... we don't want to include the header related to the cpp file first?

Reply:
Oh, it was intentional. Yes, current "defacto" code style uses:

```
<one block with stdlib headers>

<one block with external libs headers>

<one block with roc headers>
```

e.g.:

```
#include <stdio.h>
#include <stdlib.h>

#include <uv.h>

#include "roc_core/array.h"
#include "roc_core/time.h"
```